File: Examples/CSharp/ProgrammersGuide/Working-Shapes/ConfigureTimeLineShapes/RefreshTimeLine.cs
using Aspose.Diagram;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace Aspose.Diagram.Examples.CSharp.ProgrammersGuide.Working_Shapes.ConfigureTimeLineShapes
{
    public class RefreshTimeLine
    {
        public static void Run()
        {
            // ExStart:RefreshTimeLine
            // The path to the documents directory.
            string dataDir = RunExamples.GetDataDir_Shapes();

            // Load diagram
            Diagram diagram = new Diagram(dataDir + "DrawingTimeLine.vsdx");

            int shapeid = 1;
            // Get timeline shape
            Shape timeline = diagram.Pages.GetPage("Page-1").Shapes.GetShape(shapeid);

            // Initialize TimeLineHlper object
            TimeLineHelper timelineHelper = new TimeLineHelper(timeline);

            // Set start time
            timelineHelper.TimePeriodStart = new DateTime(2014, 12, 21);
            // Set end time
            timelineHelper.TimePeriodFinish = new DateTime(2015, 2, 19);

            // Set date format
            timelineHelper.DateFormatForBE = 21;

            // Revive milestones on the timeline
            timelineHelper.RefreshTimeLine();

            // Save to VDX format
            diagram.Save(dataDir + "RefreshTimeLine_Out.vsdx", SaveFileFormat.VSDX);
            // ExEnd:RefreshTimeLine
        }
    }
}

File: Examples/VisualBasic/Working-Shapes/ConfigureTimeLineShapes/RefreshTimeLine.vb
Imports Aspose.Diagram
Imports System
Public Class RefreshTimeLine
    Public Shared Sub Run()
        ' ExStart:RefreshTimeLine

        ' The path to the documents directory.
        Dim dataDir As String = RunExamples.GetDataDir_Shapes()

        ' Load diagram
        Dim diagram As New Diagram(dataDir & Convert.ToString("DrawingTimeLine.vsdx"))

        Dim shapeid As Integer = 1
        ' Get timeline shape
        Dim timeline As Shape = diagram.Pages.GetPage("Page-1").Shapes.GetShape(shapeid)

        ' Initialize TimeLineHlper object
        Dim timelineHelper As New TimeLineHelper(timeline)

        ' Set start time
        timelineHelper.TimePeriodStart = New DateTime(2014, 12, 21)
        ' Set end time
        timelineHelper.TimePeriodFinish = New DateTime(2015, 2, 19)

        ' Set date format
        timelineHelper.DateFormatForBE = 21

        'revive milestones on the timeline
        timelineHelper.RefreshTimeLine()

        ' Save to VDX format
        diagram.Save(dataDir & Convert.ToString("RefreshTimeLine_out.vsdx"), SaveFileFormat.VSDX)
        ' ExEnd:RefreshTimeLine
    End Sub
End Class
